react/components/DropdownProductQuantity.tsx

```diff
@@ -0,0 +1,169 @@
+    return ''
+  }
+
+  return `${normalizeValue(parsedValue, maxValue)}`
+}
+
+const getDropdownOptions = (maxValue: number) => {
+  const limit = Math.min(MAX_DROPDOWN_VALUE - 1, maxValue)
+  const options = []
+
+  for (let idx = 1; idx <= limit; ++idx) {
+    options.push({ value: idx, label: `${idx}` })
+  }
+
+  if (maxValue >= MAX_DROPDOWN_VALUE) {
+    options.push({
+      value: MAX_DROPDOWN_VALUE,
+      label: `${MAX_DROPDOWN_VALUE}+`,
+    })
+  }
+
+  return options
+}
+
+const CSS_HANDLES = [
+  'quantitySelectorDropdownMobileContainer',
+  'quantitySelectorDropdownContainer',
+  'quantitySelectorInputMobileContainer',
+  'quantitySelectorInputContainer',
+] as const
+
+type InternalBehavior = 'dropdown' | 'input'
+
+const DropdownProductQuantity: FunctionComponent<DropdownProps> = ({
+  itemId,
+  selectedQuantity,
+  size = 'small',
+  onChange,
+  availableQuantity,
+}) => {
+  const [behavior, setInternalBehavior] = useState<InternalBehavior>('dropdown')
+  const [curDisplayValue, setDisplayValue] = useState(`${selectedQuantity}`)
+
+  const handles = useCssHandles(CSS_HANDLES)
+  const dropdownOptions = getDropdownOptions(availableQuantity)
+
+  const handleChange = (value: string) => {
+    const validatedValue = validateValue(value, availableQuantity)
+    const displayValue = validateDisplayValue(value, availableQuantity)
+
+    if (
+      behavior === 'dropdown' &&
+      validatedValue >= Math.min(availableQuantity, MAX_DROPDOWN_VALUE)
+    ) {
+      setInternalBehavior('input')
+    }
+
+    setDisplayValue(displayValue)
+    onChange({ value: validatedValue })
+  }
+
+  const handleInputBlur = () => {
+    if (curDisplayValue === '') {
+      setDisplayValue('1')
+    }
+
+    const validatedValue = validateValue(curDisplayValue, availableQuantity)
+
+    if (validatedValue < Math.min(availableQuantity, MAX_DROPDOWN_VALUE)) {
+      setInternalBehavior('dropdown')
+    }
+
+    onChange({ value: validatedValue })
+  }
+
+  if (behavior === 'dropdown') {
+    return (
+      <Fragment>
+        <div
+          className={`${handles.quantitySelectorDropdownMobileContainer} dn-m`}>
+          <Dropdown
+            id={`quantity-dropdown-mobile-${itemId}`}
+            testId={`quantity-dropdown-mobile-${itemId}`}
+            options={dropdownOptions}
+            size={size}
+            value={selectedQuantity}
+            onChange={(event: any) => handleChange(event.target.value)}
+            placeholder=" "
+          />
+        </div>
+        <div className={`${handles.quantitySelectorDropdownContainer} dn db-m`}>
+          <Dropdown
+            id={`quantity-dropdown-${itemId}`}
+            testId={`quantity-dropdown-${itemId}`}
+            options={dropdownOptions}
+            size={size}
+            value={selectedQuantity}
+            onChange={(event: any) => handleChange(event.target.value)}
+            placeholder=" "
+          />
+        </div>
```

Comment:
Instead of always rendering two `<Dropdown />` components, you can use the `useDevice` hook from `render-runtime` to check in what device type this component is being rendered. Based on that, you can render a single `<Dropdown />`, and just change the CSS being applied